gtsfm/data_association/point3d_initializer.py

```diff
@@ -55,6 +56,54 @@ class TriangulationParam(Enum):
     RANSAC_TOPK_BASELINES = 3  # deterministically choose hypotheses with largest estimate baseline
 
 
+class TriangulationOptions(NamedTuple):
+    """Options for triangulation solver.
+
+    Refs:
+      - https://github.com/colmap/colmap/blob/dev/src/optim/ransac.h
+      - http://www.cse.psu.edu/~rtc12/CSE486/lecture15.pdf
+
+    Args:
+        reproj_error_threshold: the maximum reprojection error allowed.
+        mode: triangulation mode, which dictates whether or not to use robust estimation.
+        min_inlier_ratio: a priori assumed minimum probability that a point is an inlier.
+        confidence: desired confidence that at least one hypothesis is outlier free.
+        dyn_num_hypotheses_multiplier: multiplication factor for dynamically computed hyptheses based on confidence.
+        min_num_hypotheses: minimum number of hypotheses.
+        mas_num_hypotheses: masimum number of hypotheses.
+    """
+
+    reproj_error_threshold: float
+    mode: TriangulationParam
+
+    # RANSAC parameters
+    min_inlier_ratio: float = 0.1
+    confidence: float = 0.9999
+    dyn_num_hypotheses_multiplier: float = 3.0
+    min_num_hypotheses: int = 0
+    max_num_hypotheses: int = sys.maxsize
+
+    def __check_ransac_params(self) -> None:
+        """Check that input value are valid"""
+        assert self.reproj_error_threshold > 0
+        assert 0 < self.min_inlier_ratio < 1
+        assert 0 < self.confidence < 1
+        assert self.dyn_num_hypotheses_multiplier > 0
+        assert 0 <= self.min_num_hypotheses < self.max_num_hypotheses
+
+    def num_ransac_hypotheses(self) -> int:
+        """Compute maximum number of hypotheses."""
+        self.__check_ransac_params()
+        dyn_num_hypotheses = int(
+            np.log(1 - self.confidence)
+            / np.log(1 - self.min_inlier_ratio ** NUM_SAMPLES_PER_RANSAC_HYPOTHESIS)
+            * self.dyn_num_hypotheses_multiplier
```

Comment:
Hi Travis, could you add a reference for this dynamic multiplier? If COLMAP is using it, do you mind adding an inline comment w/ URL to the COLMAP source where they do it?

Reply:
COLMAP uses it, but all it does it multiply the number of dynamic hypotheses by a constant so I'm indifferent about whether it warrants a reference